pkg/gke/update.go

```diff
@@ -350,7 +351,7 @@ func UpdateLabels(
 	if err != nil {
 		return NotChanged, err
 	}
-	logrus.Infof("updating cluster labels for cluster [%s]", config.Name)
+	logrus.Infof("updating cluster labels (upstream: %+v; config: %+v) for cluster [%s (%s)]", upstreamSpec.Labels, config.Spec.Labels, config.Spec.ClusterName, config.Name)
```

Comment:
it is wrong, NodePool names can't be different from upstream. Using %+v as formatting verb is really bad idea.